cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java

```diff
@@ -209,74 +202,72 @@ public void resetTxManagerState(HttpRequest request, HttpResponder responder) {
     responder.sendStatus(HttpResponseStatus.OK);
   }
 
-  @Path("/transactions/pruneinfo")
-  @POST
-  public void getPruneInfo(HttpRequest request, HttpResponder responder) {
+  @Path("/transactions/prune-info/{region-name}")
+  @GET
+  public void getPruneInfo(HttpRequest request, HttpResponder responder, @PathParam("region-name") String regionName) {
     if (debugClazz == null || debugObject == null) {
       responder.sendString(HttpResponseStatus.BAD_REQUEST, "Invalid List Pruning debug class not found.");
       return;
     }
 
-    try (Reader reader = new InputStreamReader(new ChannelBufferInputStream(request.getContent()))) {
-      String regionName = CharStreams.toString(reader);
+    try {
       Method method = debugClazz.getMethod("getRegionPruneInfo", String.class);
       method.setAccessible(true);
       Object response = method.invoke(debugObject, regionName);
       if (response == null) {
-        responder.sendString(HttpResponseStatus.OK, "No prune upper bound has been registered for this region yet.");
+        responder.sendString(HttpResponseStatus.NOT_FOUND,
+                             "No prune upper bound has been registered for this region yet.");
         return;
       }
       RegionPruneInfo pruneInfo = (RegionPruneInfo) response;
-      responder.sendString(HttpResponseStatus.OK, GSON.toJson(pruneInfo));
+      responder.sendJson(HttpResponseStatus.OK, pruneInfo);
     } catch (Exception e) {
       responder.sendString(HttpResponseStatus.BAD_REQUEST, e.getMessage());
       LOG.debug("Exception while trying to fetch the RegionPruneInfo.", e);
     }
   }
 
-  @Path("/transactions/timeregions")
-  @POST
-  public void getTimeRegions(HttpRequest request, HttpResponder responder) {
+  @Path("/transactions/time-regions/{time-stamp}")
```

Comment:
This one can be `/transactions/prune/regions?time={timestamp}`

@awholegunch What is the recommended name for timestamp parameter?